data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/external/ExternalEvent.avsc

```diff
@@ -0,0 +1,31 @@
+{
+  "type": "record",
+  "name": "ExternalEvent",
+  "namespace": "net.corda.data.flow.event.external",
+  "doc": "Events sent to workers that are triggered from flow APIs",
+  "fields": [
+    {
+      "name": "topic",
+      "type": "string",
+      "doc": "The topic that the external event is sent to."
+    },
+    {
+      "name": "key",
+      "type": "bytes",
+      "doc": "Avro serialized representation of the key that the external event is partitioned by."
+    },
+    {
+      "name": "payload",
+      "type": "bytes",
```

Comment:
I raised this JIRA a little while ago: https://r3-cev.atlassian.net/browse/CORE-5931

Given we are refactoring this, would it be worth introducing this now?
I think for now the AMQPPayload schema could probably just be a wrapper for `bytes`, but would make the API more explicit.

Reply:
In this case it would be an `AvroPayload` though rather than `AMQPPayload`, just making sure we're on the same page.

Reply:
We changed this deliberately. The problem with this being a union is that we're expecting to have to generate external events from layers of the API higher up than the flow API (ledger being the primary example). We're going to rapidly run into an extensibility problem here if we force a union where we don't control all the possible variants that could be in that union.

Reply:
Was thinking about this again this morning, not sure why, but shouldn't we use [the existing `AvroEnvelope`](https://github.com/corda/corda-api/blob/release/os/5.0/data/avro-schema/src/main/resources/avro/net/corda/data/AvroEnvelope.avsc) in this case?
@lankydan , @JamesHR3 

I'm thinking again in case you want to archive Kafka messages for analysis, or even just monitoring (some orgs will want to do this). From building [KafkaSnoop](https://github.com/driessamyn/kafkasnoop), I know you need to have some indication of what the schema is in order to do anything with the binary payload. 
With AMQP it's less important as you don't need a schema (although you still will want to know it's AMQP).

Reply:
@driessamyn @JamesHR3 I'll raise a tech debt ticket for this and address it separately.

Reply:
Created - https://r3-cev.atlassian.net/browse/CORE-6401

Reply:
The serialization library is reading the schema from the byte array which contains the fingerprint. So the ByteArray is a lot more self-describing than it seems.